src/modules/System/Api/Guest.php

```diff
@@ -23,6 +23,19 @@ class Guest extends \Api_Abstract
      */
     public function version()
     {
+        // check if the user is logged in as admin and if so, return the version
+        if ($this->di['auth']->isAdminLoggedIn()) {
+            return $this->getService()->getVersion();
+        }
+
+        // check if the "show_version_public" parameter is set to true
+        $showVersionPublic = $this->getService()->getParamValue('show_version_public');
+        if ($showVersionPublic == 1) {
```

Comment:
Don't need to check and verify if it's set to 1.
`if($showVersionPublic)` will evaluate to true if the value is set and non-falsey. Meaning if it's set and is equal to 1, the check will pass. 

If it's not set or it's set to 0, the check will not pass.